web/src/actions/auth.js

```diff
@@ -25,6 +26,18 @@ export const failLogin = error => ({ type: LOGIN_FAILURE, error });
 
 export const completeLogout = () => ({ type: LOGOUT_SUCCESS });
 
+const loadData = activities => dispatch => {
+  if (activities.includes('view-document')) {
+    dispatch(fetchApd());
+  }
+  if (activities.includes('view-users')) {
+    dispatch(getUsers());
+  }
+  if (activities.includes('view-roles')) {
+    dispatch(getRoles());
+  }
+};
```

Comment:
Check the user's permissions to see which data to load, and dispatch those loads.